include/ensmallen_bits/adam/qhadam_update.hpp

```diff
@@ -0,0 +1,156 @@
+/**
+ * @file qhadam_update.hpp
+ * @author Niteya Shah
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_ADAM_QHADAM_UPDATE_HPP
+#define ENSMALLEN_ADAM_QHADAM_UPDATE_HPP
+
+namespace ens {
+
+/**
+ *
+ * TODO: Fill in these details as well on info about algorithm
+ * For more information, see the following.
+ *
+ * @code
+ * @article{Kingma2014,
```

Comment:
I think we should cite the Quasi-hyperbolic momentum and Adam for deep learning paper here.